pytransform3d/trajectories.py

```diff
@@ -436,6 +509,201 @@ def pqs_from_dual_quaternions(dqs):
+    -------
+    qs : array, shape (…,3)
+        Vector to a point on the screw axis
+
+    s_axiss : array, shape (…,3)
+        Direction vector of the screw axis
+
+    hs : array, shape (…,)
+        Pitch of the screw. The pitch is the ratio of translation and rotation
+        of the screw axis. Infinite pitch indicates pure translation.
+
+    thetas : array, shape (…,)
+        Parameter of the transformation: theta is the angle of rotation
+        and h * theta the translation.
+    """
+    reals = dqs[:, :4]
+    duals = dqs[:, 4:]
+
+    # Vectorized axis angle from quaternion
+    a = axis_angles_from_quaternions(reals)  # Should handle vectorized input
+    s_axis = a[:, :3]
+    thetas = a[:, 3]
+
+    # Vectorized quaternion concatenation and translation computation
+    translation = 2 * batch_concatenate_quaternions(duals, batch_q_conj(reals))[:, 1:]
+
+
+    outer_if_mask = np.abs(thetas) < np.finfo(float).eps
+    outer_else_mask = np.logical_not(outer_if_mask)
+
+
+    ds = np.linalg.norm(translation, axis=1)
+    inner_if_mask = ds < np.finfo(float).eps
+
+    outer_if_inner_if_mask = np.logical_and(outer_if_mask,inner_if_mask)
+    outer_if_inner_else_mask = np.logical_and(outer_if_mask,np.logical_not(inner_if_mask))
+    
+    # Initialize the outputs
+    qs = np.zeros((dqs.shape[0], 3))  
+    thetas[outer_if_mask] = ds[outer_if_mask]
+    hs = np.full(dqs.shape[0], np.inf)  
+    
+    if np.any(outer_if_inner_if_mask):
+        s_axis[outer_if_inner_if_mask] = np.array([1, 0, 0])
+
+    if np.any(outer_if_inner_else_mask):
+        s_axis[outer_if_inner_else_mask] = translation[outer_if_inner_else_mask] / ds[outer_if_inner_else_mask]
+
+    qs = np.zeros((dqs.shape[0], 3))  
+    thetas[outer_if_mask] = ds[outer_if_mask]
+    hs = np.full(dqs.shape[0], np.inf)  
+    
+    if np.any(outer_else_mask):
+        distance = np.einsum('ij,ij->i', translation[outer_else_mask], s_axis[outer_else_mask])
+
+        moment = 0.5 * (
+            np.cross(translation[outer_else_mask], s_axis[outer_else_mask]) +
+            (translation[outer_else_mask] - distance[:, None] * s_axis[outer_else_mask])
+            / np.tan(0.5 * thetas[outer_else_mask])[:, None]
+        )
+
+        qs[outer_else_mask] = np.cross(s_axis[outer_else_mask], moment)
+        hs[outer_else_mask] = distance / thetas[outer_else_mask]
+
+    return qs, s_axis, hs, thetas
+
+def dual_quaternions_from_screw_parameters(qs, s_axis, hs, thetas):
+    ds = np.zeros_like(hs)
+    
+    h_is_inf_mask = np.isinf(hs)
+    h_is_not_inf_mask = np.logical_not(h_is_inf_mask)
+
+    mod_thetas = thetas.copy()
+    if np.any(h_is_inf_mask):
+        ds[h_is_inf_mask] = thetas[h_is_inf_mask]
+        mod_thetas[h_is_inf_mask] = 0
+    
+    if np.any(h_is_not_inf_mask):
+        ds[h_is_not_inf_mask] = hs[h_is_not_inf_mask] * thetas[h_is_not_inf_mask]
+
+    moments = np.cross(qs, s_axis)
+    half_distances = 0.5 * ds
+    sin_half_angles = np.sin(0.5 * mod_thetas)
```

Comment:
`0.5 * mod_thetas` could be stored in a temporary variable. Since this function could be applied to large arrays, we should do this.